service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -288,34 +292,38 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 	return entryCRs, nil
 }
 
-func sortEntry(entries []entry) ([]entry, error) {
-	if len(entries) == 0 {
-		return nil, nil
-	}
+func (r *Resource) getLatestVersion(ctx context.Context, entries []entry) (string, error) {
```

Comment:
This is the latest version by semver right? Can you add godoc for that so its clear its not the latest by create date.